ckan/install.sls

```diff
@@ -35,28 +35,25 @@ ckan-user:
 ckan-venv:
   pkg.installed:
     - pkgs:
-      - python-virtualenv
       - {{ ckan.python_pip }}
 
+  cmd.run:
+    - name: pip install --user virtualenv
+    - user: {{ ckan.ckan_user }}
+    - creates: {{ home }}/.local/bin/virtualenv
```

Comment:
use `pip.installed` and pass `upgrade: true`

Reply:
Unfortunately, seems that salt pip.installed has no equivalent for `--user`.

https://docs.saltstack.com/en/latest/ref/states/all/salt.states.pip_state.html#module-salt.states.pip_state